src/blockchain.rs

```diff
@@ -0,0 +1,181 @@
+use crate::block::Block;
+use std::collections::HashMap;
+
+/// Indexes of chain attribute
+#[derive(Debug, Clone)]
+pub struct BlockchainIndex {
```

Comment:
I don't think we can possibly keep this all in memory... If we need a way to access blocks from deep down the chain we'll need to make some sort of lazy-loading cache or something...

Reply:
Maybe it's okay to keep this around for now, btu we need an interface to this structure that's going to allow us to store these on disk...

Reply:
LGTM